### PR TITLE
xDS cluster resource type: use ValidationErrors and add unit test

### PR DESCRIPTION
**Pull request by markdroth**
Also adds some missing validation for outlier detection parameters, as per [gRFC A50](https://github.com/grpc/proposal/blob/master/A50-xds-outlier-detection.md#validation).  I added this both to the xDS resource validation and to the LB policy config parsing code, with appropriate tests.

### CODE REVIEW

File: src/core/ext/xds/xds_common_types.cc [hunk 1]
```diff
@@ -45,6 +46,25 @@
...
 namespace grpc_core {
 
+//
+// ParseDuration()
+//
+
+Duration ParseDuration(const google_protobuf_Duration* proto_duration,
+                       ValidationErrors* errors) {
+  int64_t seconds = google_protobuf_Duration_seconds(proto_duration);
+  if (seconds < 0 || seconds > 315576000000) {
```
Comment (yashykt) [inline src/core/ext/xds/xds_common_types.cc:56]: we are using these constants at multiple places. Let's define them in one place and use that name everywhere

**Review by yashykt — COMMENTED** [review]
I seem to be missing something on how ScopedField works

**Review by yashykt — APPROVED** [review]
I don't have any objections to this structure but it does seem like in some places instead of a single return we have more extra lines

File: src/core/ext/xds/xds_common_types.cc (hunk 1 above)
Comment (markdroth) [reply src/core/ext/xds/xds_common_types.cc:56]: We're using them in exactly two places, here and in the JSON parsing code.  I don't really see a convenient common dependency where we could put the constant, and I don't think the duplication is going to cause enough of a problem to warrant the overhead of creating a new module just for this constant.

**Pull request merged**